mantidproject/mantid
# User-defined background subtraction algorithm

`CreateUserDefinedBackground` takes a set of points selected by the user in the GUI and returns a
workspace that can be subtracted from the data. The intended use case is described in the issue.

**To test:**
- Create or load a workspace that has a large, irregularly shaped background with several peaks on top of it.
- In MantidPlot, use _Data/Draw Data Points_ to define the background by double-clicking on a series of points. When you click outside of the graph window, a table will appear with your points in them.
  - (On Linux the table won't appear until you close the graph window - see #16874)
- With the table selected, convert to a workspace via _Table/Convert to TableWorkspace_
- Run the algorithm:

``` python
background = CreateUserDefinedBackground(InputWorkspace = "your input", BackgroundPoints = "your table")
```
- Ensure that the resulting workspace `background` can be subtracted from the original data workspace (using `Minus`) and the result contains just the peaks.

Fixes #16798.

[Release notes](https://github.com/mantidproject/mantid/blob/37607a4d6e6dd762cdec4b562170dcc628c2a5c1/docs/source/release/v3.8.0/framework.rst#new) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
